builder/processor/src/main/java/io/helidon/builder/processor/CustomMethods.java

```diff
@@ -0,0 +1,284 @@
+                annotations,
+                generatedCall);
+    }
+
+    private static boolean correctType(TypeName knownType, TypeName processingType) {
+        // processing type may be for a generated class, which does not contain package information
+        if (processingType.packageName().isEmpty()) {
+            if (processingType.className().equals("<any>")) {
+                // cannot be resolved as this is part of our round, good faith it is a correct parameters
+                return true;
+            }
+            return knownType.className().equals(processingType.className())
+                    && knownType.enclosingNames().equals(processingType.enclosingNames());
+        }
+        return knownType.equals(processingType);
+    }
+
+    // static methods on prototype
+    private static GeneratedMethod factoryMethod(Errors.Collector errors,
+                                                 TypeContext.TypeInformation typeInformation,
+                                                 TypeName customMethodsType,
+                                                 List<String> annotations,
+                                                 Method customMethod) {
+
+        // if void: CustomMethodsType.methodName(param1, param2)
+        // if returns: return CustomMethodsType.methodName(param1, param2)
+        String generatedCall =
+                VOID_TYPE.equals(customMethod.returnType) ? "" : "return "
+                        + customMethodsType.genericTypeName().fqName() + "."
+                        + customMethod.name()
+                        + "("
+                        + customMethod.arguments().stream().map(Argument::name).collect(Collectors.joining(", "))
+                        + ")";
+
+        // factory methods are just copied to the generated prototype
+        return new GeneratedMethod(new Method(typeInformation.prototype(),
+                                              customMethod.name(),
+                                              customMethod.returnType(),
+                                              customMethod.arguments(),
+                                              customMethod.javadoc()),
+                                   annotations,
+                                   generatedCall);
+    }
+
+    private static List<CustomMethod> findMethods(TypeContext.TypeInformation typeInformation,
+                                                  TypeInfo customMethodsType,
+                                                  Errors.Collector errors,
+                                                  TypeName requiredAnnotation,
+                                                  MethodProcessor methodProcessor) {
+        // all custom methods must be static
+        // parameter and return type validation is to be done by method processor
+        return customMethodsType.elementInfo()
+                .stream()
+                .filter(TypeInfoPredicates::isMethod)
```

Comment:
more `is` methods??

Reply:
What does your comment even mean?

Reply:
I was under the belief that `is`XXX methods were frowned upon from a style guide perspective.

Reply:
The method is a predicate - the `is` prefix makes sense